tools/deb/istio.mk

```diff
@@ -0,0 +1,83 @@
+.PHONY: docker.istio.deb istio.deb sidecar.deb ${OUT}/istio-sidecar.deb
+
+# Make the deb image using the CI/CD image and docker.
+docker.istio.deb:
+	(cd ${TOP}; docker run --rm -u $(shell id -u) -it \
+        -v ${GO_TOP}:${GO_TOP} \
+        -w ${PWD} \
+        -e USER=${USER} \
+        -e GOPATH=${GOPATH} \
+		--entrypoint /usr/bin/make ${CI_HUB}/ci:${CI_VERSION} \
+		sidecar.deb )
+
+# Create the 'sidecar' deb, including envoy and istio agents and configs.
+# This target uses a locally installed 'fpm' - use 'docker.sidecar.deb' to use
+# the builder image.
+# TODO: consistent layout, possibly /opt/istio-VER/...
+sidecar.deb: ${OUT}/istio-sidecar.deb
+
+istio.deb: ${OUT}/istio-sidecar.deb
+
+ISTIO_DEB_SRC=tools/deb/istio-start.sh \
+			  tools/deb/istio-iptables.sh \
+			  tools/deb/istio.service \
+			  tools/deb/istio-auth-node-agent.service \
+			  tools/deb/sidecar.env \
+			  tools/deb/envoy.json
+
+ISTIO_DEB_DEPS=${ISTIO_BIN}/envoy \
+			   ${ISTIO_BIN}/pilot-agent \
+			   ${ISTIO_BIN}/node_agent \
+			   ${ISTIO_BIN}/istioctl \
+			   ${ISTIO_BIN}/mixs \
+			   ${ISTIO_BIN}/istio_ca
```

Comment:
Looks like ${ISTIO_BIN}/pilot-discovery is missing.

Reply:
Good catch. If someone knows some magic to generate the source=${ISTIO_DEB_BIN}/lastpart we can improve this a bit.

Reply:
If I had to venture a guess it'd be:
ISTIO_DEB_DEPS:=envoy pilot-agent node_agent istioctl mixs istio_ca

\# add depdencies for istio-sidecar.deb on the istio binaries it consumes
$(foreach DEP,$(ISTIO_DEB_DEPS),$(eval ${OUT}/istio-sidecar.deb: ${ISTIO_BIN}/$(DEP)))

\# generate a list of sources for the istio binaries
ISTIO_DEB_DEPS_SOURCES:=
$(foreach DEP,$(ISTIO_DEB_DEPS),$(eval ISTIO_DEB_DEPS_SOURCES+=bin/$(DEP)=${ISTIO_DEB_BIN}/$(DEP)))

and then pass the $(ISTIO_DEB_DEPS_SOURCES) to fpm.